### DISCUSSION TITLE
"Group Metrics" not working in Alerts after 2.4.80 Update

### DISCUSSION THREAD

**Discussion by brazorfajeje** [Security-Onion-Solutions/securityonion#13327] [2.4] [answered]
### Version

2.4.80

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

16

### RAM

32

### Storage for /

250GB

### Storage for /nsm

1TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

After updating to 2.4.80 I can no longer see grouped results in  "Alerts" in SOC.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/92110885/eb724c1a-02cb-463b-9b65-f3000dfe208f)

The same query in the Hunts works correctly

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by defensivedepth** [comment]
Put an asterisk after the last two fields and then post a screenshot please

**Reply by brazorfajeje** [reply]
do you have any advice on what I could check to see why the metrics are not being calculated correctly?

**Comment by brazorfajeje** [comment]
Thanks,

same result:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/92110885/4e58936b-1412-4630-87ea-9a440e5dfd0a)

tryied with askerisk afeter rule.name and got this:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/92110885/ffc8db1f-d9d0-427e-9d6a-e6e225ee7885)


in the Alerts section I have data and all those fields are correctly populated

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/92110885/acffb5d3-b59b-47b3-871a-059c1c7de966)

**Comment by brazorfajeje** [comment]
can you recommend what logs I could analyze to try to figure out where the problem is?

**Comment by defensivedepth** [comment]
The alerts that are showing missing, but showing the `rule.uuid` - Filter for one of those and post it here.

**Comment by brazorfajeje** [comment]
![image](https://github.com/user-attachments/assets/677ba402-ee45-439d-86aa-36ebc22d0180)


thanks

**Comment by brazorfajeje** [comment]
is it possible that something changed in the last update regarding some configuration file (something in /opt/so/saltstack/local/pillar/soc/ for example) and therefore a custom configuration was not properly updated?

The data as from the screenshot above is present, the problem is only in the aggregate data view which is missing.

**Comment by brazorfajeje** [comment]
since the server correctly responds to the api call to /api/events/ with a JSON containing all the fields (as then you can also see from the last screenshot the data is present) I guess it is a problem of the js that is in charge of compiling the table.
From https://github.com/Security-Onion-Solutions/securityonion-soc/pull/569 I have not been able to identify what might have changed the functioning of soc and thus caused the problem.
Can you at least point me to which js it might make sense to audit?

thanks

--- EDIT---
if it helps I also add that in the browser console I have the following warnings:
```
Source map error: request failed with status 404
Resource URL: https://<so>/js/external/vue-prism-editor-v1.3.0.umd.production.min.js
Source Map URL: prismeditor.umd.production.min.js.map
```

```
downloadable font: Glyph bbox was incorrect (glyph ids 2 3 5 8 9 10 11 12 13 14 16 17 19 22 24 32 34 35 38 39 40 43 44 45 46 47 50 51 52 53 55 56 58 60 61 62 64 67 68 70 71 72 73 74 78 79 80 81 83 90 96 101 103 104 105 108 109 115 116 117 118 120 123 125 134 137 138 140 141 142 143 144 146 147 152 153 155 159 160 161 162 163 167 168 171 177 191 193 201 205 206 208 209 212 216 217 221 223 225 226 227 228 233 234 235 236 237 238 243 244 245 246 247 248 249 250 251 252 253 254 255 261 262 264 266 269 273 276 277 278 279 281 284 285 286 289 290 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 319 333 336 337 338 340 342 343 344 351 352 354 355 356 357 360 361 363 364 369 372 377 379 380 381 384 387 388 389 391 392 404 405 410 411 414 416 417 418 422 430 431 437 446 447 448 449 452 453 454 455 470 477 478 479 480 483 484 488 489 491 497 498 499 501 506 507 513 514 523 525 526 530 533 540 541 547 548 549 550 552 553 554 556 558 567 569 598 600 601 602 605 607 612 613 615 616 621 635 641 642 643 645 648 649 652 653 654 655 660 661 662 663 668 669 670 672 673 677 678 679 680 681 696 697 706 712 715 716 727 728 730 734 737 744 745 750 759 760 772 774 775 776 777 786 787 788 792 794 796 798 799 801 802 804 824 826 827 829 833 834 836 837 838 839 840 841 842 843 846 847 854 855 859 869 871 872 878 880 887 893 896 898 906 909 911 923 926 927 928 931 934 935 936 939 940 941 942 943 946 947 948 950 956 958 959 960 962 964 965 967 971 972 976 977 978 979 980 981 987 998 999 1003 1004 1006 1007 1009 1010 1011 1014 1018 1024 1025 1029 1034 1035 1040 1043 1046 1048 1050 1051 1055 1056 1058 1061 1070 1071 1074 1082 1085 1102 1108 1109 1110 1111 1114 1115 1119 1120 1122 1129 1131 1134 1138 1139 1140 1145 1146 1149 1155 1161 1165 1166 1168 1175 1184 1185 1191 1194 1197 1198 1199 1202 1203 1206 1209 1210 1215 1216 1218 1222 1224 1226 1227 1228 1229 1230 1231 1233 1234 1235 1236 1241 1244 1245 1247 1249 1254 1256 1257 1258 1259 1263 1267 1269 1270 1271 1272 1273 1274 1277 1283 1287 1288 1289 1290 1294 1295 1301 1302 1303 1307 1308 1309 1310 1315 1317 1318 1322 1323 1326 1327 1328 1329 1332 1333 1335 1337 1339 1341 1343 1354 1355 1361 1367 1369 1370 1373 1374 1375 1380 1382 1383 1385) (font-family: "Font Awesome 6 Free" style:normal weight:900 stretch:100 src index:0) source: https://<so>/css/webfonts/fa-solid-900.woff2
```

```
Some cookies are misusing the recommended “SameSite“ attribute 4
```

**Comment by brazorfajeje** [comment]
Debugging the behavior of hunt.js I just noticed that metrics are computed server-side and sent to the client in the metrics field. i confirm that these metrics result `__missing__` in the json response to the api call

here a snippet of the response:
```
    "metrics":
    {
        "bottom":
        [],
        "groupby_0|rule.name":
        [
            {
                "keys":
                [
                    "__missing__"
                ],
                "value": 4
            }
        ],
        "groupby_0|rule.name|event.module":
        [
            {
                "keys":
                [
                    "__missing__",
                    "__missing__"
                ],
                "value": 4
            }
        ],
        "groupby_0|rule.name|event.module|event.severity_label":
        [
            {
                "keys":
                [
                    "__missing__",
                    "__missing__",
                    "__missing__"
                ],
                "value": 4
            }
        ],
        "groupby_0|rule.name|event.module|event.severity_label|rule.uuid":
        [
            {
                "keys":
                [
                    "__missing__",
                    "__missing__",
                    "__missing__",
                    "2010935"
                ],
                "value": 2
            },
            {
                "keys":
                [
                    "__missing__",
                    "__missing__",
                    "__missing__",
                    "2402000"
                ],
                "value": 1
            },
            {
                "keys":
                [
                    "__missing__",
                    "__missing__",
                    "__missing__",
                    "2500010"
                ],
                "value": 1
            }
        ],
        "timeline":
        [
            {
                "keys":
                [
                    "2024-07-22T13:47:06.000Z"
                ],
                "value": 2
            },
            {
                "keys":
                [
                    "2024-07-22T13:47:21.000Z"
                ],
                "value": 1
            },
            {
                "keys":
                [
                    "2024-07-22T13:47:25.000Z"
                ],
                "value": 1
            }
        ]
    }
    ``

**Comment by brazorfajeje** [comment]
@defensivedepth could you help me figure out which component is responsible for calculating the metrics and therefore generating this missing value?  if so, I could try to look at the logic and try to see if there is something wrong with my indexes

thanks in advance

**Comment by brazorfajeje** [comment, marked as answer]
The issue magically resolved itself (probably after all the pre-update suricata indices have been deleted by ILM)  :grimacing: 
Mark as closed
